python/ray/air/util/tensor_extensions/pandas.py

```diff
@@ -1361,3 +1384,44 @@ def _is_boolean(self):
 TensorArray._add_arithmetic_ops()
 TensorArray._add_comparison_ops()
 TensorArray._add_logical_ops()
+
+
+@PublicAPI(stability="beta")
+def is_ndarray_variable_shaped_tensor(arr: np.ndarray) -> bool:
```

Comment:
This is heavily tied to the internal variable-shaped tensor representation of the Pandas-side of the tensor extension type (which uses a NumPy ndarray for its internal representation), so I was planning on housing it next to the extension type since (1) it uses it internally to determine whether its variable-shaped or not, and (2) we need this utility and the `TensorDtype` and `TensorArray` representations to stay in sync.

Reply:
I opted to move it to a `tensor_extensions/utils.py` so both `ArrowTensorArray` and `TensorArray` can use it and we can hide it from the user.

Reply:
>I opted to move it to a tensor_extensions/utils.py so both ArrowTensorArray and TensorArray can use it and we can hide it from the user.

Yeah, this sounds good to me.